sympy/physics/units/util.py

```diff
@@ -154,6 +158,46 @@ def quantity_simplify(expr):
         ref = v[0]/v[0].scale_factor
         expr = expr.xreplace({vi: ref*vi.scale_factor for vi in v[1:]})
 
+    if across_dimensions:
+        # combine quantities of different dimensions into a single
+        # quantity that is equivalent to the original expression
+
+        unit_system = UnitSystem.get_unit_system("SI")
```

Comment:
What about if you are not using SI unit systems?

Reply:
Ah, yeah I forgot to fix that

Reply:
I've added an extra argument for the unit system. However, I am uncertain on whether or not this sufficiently solves the problem because of the hard coded dict `preferred_dimension_to_unit`. 

Realistically though, I don't think we should consider it a blocker for merging because in 99.99% of cases, people are going to be using SI.